### DISCUSSION TITLE
Run individual unit test maven

### DISCUSSION THREAD

**Discussion by csrcordeiro** [apache/netbeans#8048] [Q&A] [answered]
In a maven based project, is it possible to run individual JUnit tests without going through the maven build process?

**Comment by matthiasblaesing** [comment, marked as answer]
In general you can run individual tests by invoking "Test File" from the context menu of the test class. That will invoke the maven build and run that single test. You can try to use "Compile on save" (Project settings: Build -> Compile):

![grafik](https://github.com/user-attachments/assets/9b6a0d76-f118-43fa-8d26-37aadc49630b)

BUT(!) this falls into the category "From here on dragons". If it works, it is great, but it is not without problems so try it, but if it does not work, well....

**Reply by csrcordeiro** [reply]
Ok. I'll give it a try to see if it speed things up.

Thanks
